Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -98,6 +99,56 @@ await container.ReadItemAsync<TItem>(id, partitionKey, cancellationToken: cancel
             return results;
         }
 
+        /// <inheritdoc/>
+        public async Task<PagedResult<TItem>> GetListAsync<TPagedRequest>(
+            Expression<Func<TItem, bool>> predicate, TPagedRequest pagedResultRequest,
+            CancellationToken cancellationToken = default)
+        {
+            Container container =
+                await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            IQueryable<TItem> query =
+                container.GetItemLinqQueryable<TItem>()
+                    .Where(predicate.Compose(
+                        item => !item.Type.IsDefined() || item.Type == typeof(TItem).Name, Expression.AndAlso));
+
+            Response<int> response = await query.CountAsync(cancellationToken);
+            int totalCount = response.Resource;
+
+            query = ApplyPaging(query, pagedResultRequest);
+
+            TryLogDebugDetails(_logger, () => $"Read: {query}");
+
+            using FeedIterator<TItem> iterator = query.ToFeedIterator();
+            List<TItem> results = new();
+            while (iterator.HasMoreResults)
+            {
+                foreach (TItem result in await iterator.ReadNextAsync(cancellationToken).ConfigureAwait(false))
+                {
+                    results.Add(result);
+                }
+            }
+            return new PagedResult<TItem>(totalCount, results);
+        }
+
+        protected virtual IQueryable<TItem> ApplyPaging<TPagedRequest>(IQueryable<TItem> query, TPagedRequest pagedResultRequest)

```

Comment:
Why is this `protected virtual`? What is the plan for this?

Reply:
Maybe someone can override the method to add customize pagination.